src/shared/conf-parser.c

```diff
@@ -316,10 +316,12 @@ int config_parse(const char *unit,
         fd_warn_permissions(filename, fileno(f));
 
         for (;;) {
-                char buf[LINE_MAX], *l, *p, *c = NULL, *e;
+                _cleanup_free_ char *buf = NULL;
+                size_t buf_size = 0;
+                char *l, *p, *c = NULL, *e;
                 bool escaped = false;
 
-                if (!fgets(buf, sizeof buf, f)) {
+                if (getline(&buf, &buf_size, f) < 0) {
```

Comment:
I don't grok the getline() API I must say... will it allocate unbounded amounts of memory if you point it to a large file or stream that doesn't contain newlines? (let's say /dev/urandom)

I mean, it's definitely worth discussing changing the line limit, but I am not sure we should remove it altogether, as reading unbounded amounts of input into unbounded allocated memory is simply a security problem...

ideally we had an API that works like getline(), i.e. dynamically allocates a buffer, but also enforces a maximum line length of some value we specify (could be something huge such as 16M or so), and grows the buffer dynamically.